coopr-docs/docs/source/guide/superadmin/plugins.rst

```diff
@@ -155,7 +155,21 @@ The only required return value is ``status``, where ``'status': 0`` represents s
 represents failure. A raised exception will also result in failure.
 
 Additionally, your provider plugin will likely need to return information such as a machine's ID, SSH credentials and
-public IP, so that it can be used in subsequent tasks. For these cases, simply write the results as key-value pairs underneath
+public IP, so that it can be used in subsequent tasks.  Special facilities exist for reporting the machine's hostname and
+IP addresses, so that they can be reference by :doc:`macros </guide/admin/macros>`.  To report a machine's hostname or
+IP addresses, populate the following fields:
+::
+
+  @result['hostname'] = 'some.provider.generated.hostname'
+  @result['ipaddresses'] = {
+    'access_v4' => 'x.x.x.x',
+    'bind_v4' => '10.x.x.x'
+  }
+
+where ``access_v4`` is the machine's publicly routable IP address and ``bind_v4`` is the machine's private, internal
+interface.  These entries are all optional.
```

Comment:
Does "these entries" refer to everything in the code sample, or just the access_v4 and bind_v4? If the latter, I would recommend changing to:
```
These last two entries are optional.
```